src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePumpSettings.cs

```diff
@@ -0,0 +1,102 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Arcus.Security.Core;
+using Arcus.Security.Core.Caching;
+using GuardNet;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Arcus.Messaging.Pumps.ServiceBus
+{
+    /// <summary>
+    ///     Settings for an Azure Service Bus message pump
+    /// </summary>
+    public class AzureServiceBusMessagePumpSettings
+    {
+        private readonly Func<ISecretProvider, Task<string>> _getConnectionStringFromSecretFunc;
+        private readonly Func<IConfiguration, string> _getConnectionStringFromConfigurationFunc;
+        private readonly IServiceProvider _serviceProvider;
+
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="entityName">Name of the entity to process</param>
+        /// <param name="subscriptionName">Name of the subscription to process</param>
+        /// <param name="getConnectionStringFromConfigurationFunc">Function to look up the connection string from the configuration</param>
+        /// <param name="getConnectionStringFromSecretFunc">Function to look up the connection string from the secret store</param>
+        /// <param name="options">Options that influence the behavior of the message pump</param>
+        /// <param name="serviceProvider">Collection of services to use</param>
+        public AzureServiceBusMessagePumpSettings(string entityName, string subscriptionName,
+            Func<IConfiguration, string> getConnectionStringFromConfigurationFunc,
+            Func<ISecretProvider, Task<string>> getConnectionStringFromSecretFunc,
+            AzureServiceBusMessagePumpOptions options, IServiceProvider serviceProvider)
+        {
+            Guard.For<ArgumentException>(
+                () => getConnectionStringFromConfigurationFunc == null && getConnectionStringFromSecretFunc == null,
+                "Unable to determine connection string as it was not defined how to look it up");
+            Guard.NotNull(options, nameof(options));
+            Guard.NotNull(serviceProvider, nameof(serviceProvider));
+
+            _serviceProvider = serviceProvider;
+            _getConnectionStringFromConfigurationFunc = getConnectionStringFromConfigurationFunc;
+            _getConnectionStringFromSecretFunc = getConnectionStringFromSecretFunc;
+
+            EntityName = entityName;
+            SubscriptionName = subscriptionName;
+            Options = options;
+        }
+
+        /// <summary>
+        ///     Name of the entity to process
+        /// </summary>
+        /// <remarks>This is optional as the connection string can contain the entity name</remarks>
+        public string EntityName { get; }
+
+        /// <summary>
+        ///     Name of the subscription to process
+        /// </summary>
+        /// <remarks>This is only applicable when using Azure Service Bus Topics</remarks>
+        public string SubscriptionName { get; }
```

Comment:
This is actually a cross-PR question for https://github.com/arcus-azure/arcus.webapi/pull/101
If the background job should alter the subscription name with the `JobId`; where should this happen?
With the current setup, it would mean that it has to be set in the `IServicesCollectionExtensions` and then be stripped of the subscription name, afterwards?

Reply:
Or should I introduce some options to override the JobId.

Reply:
I'll fix that, just mentioning it here.